toqito/state_opt/bell_inequality_max.py

```diff
@@ -0,0 +1,149 @@
+"""Computes the upper bound for a given bipartite Bell inequality."""
+
+from itertools import combinations
+
+import cvxpy as cp
+import numpy as np
+
+from toqito.channels import partial_transpose
+from toqito.perms import permutation_operator, swap
+
+
+def bell_inequality_max(
+    joint_coe: np.ndarray,
+    a_coe: np.ndarray,
+    b_coe: np.ndarray,
+    a_val: np.ndarray,
+    b_val: np.ndarray,
+    solver_name: str = "SCS",
+) -> float:
+    r"""Return the upper bound for the maximum violation(Tsirelson Bound) for a given bipartite Bell inequality.
+
+    This computes the upper bound for the maximum value of a given bipartite Bell inequality using an SDP.
+    The method is from :cite:`Navascues_2014_Characterization` and the implementation is based on :cite:`QETLAB_link`.
+    This is useful for various tasks in device independent quantum information processing.
+
+    The function formulates the problem as a SDP problem in the following format
+
+    .. math::
+
+        \begin{multline}
+        \max \operatorname{tr}\!\Bigl( W \cdot \sum_{a,b,x,y} B^{xy}_{ab}\, M^x_a \otimes N^y_b \Bigr),\\[1ex]
+        \text{s.t.} \quad \operatorname{tr}(W) = 1,\quad W \ge 0,\\[1ex]
+        W^{T_P} \ge 0,\quad \text{for all bipartitions } P.
+        \end{multline}
+
+
+    Examples
+    =======
+
+
+    Consider the I3322 Bell inequality
```

Comment:
Add a reference for this inequality. I don't think the one I have provided below is the original paper. 

https://arxiv.org/abs/quant-ph/0306129

You can get the bib entry by clicking on 

![image](https://github.com/user-attachments/assets/fa4a7ef9-e67b-4b56-9f8d-acafdd5dec3a)